### ISSUE TITLE
Better data handling

### ISSUE THREAD

**Issue by bjorn3** [rust-lang/rustc_codegen_gcc#119]
* [ ] Don't export non-public statics
* [ ] Don't export anonymous allocations
* [ ] Mark allocations as read-only if they are read-only

These will likely improve the binary size and will help with optimizations.

**Comment by antoyo** [comment]
That's a good idea.
Maybe avoiding exporting will not help with binary size because of the linker script, but it's still worth a try!

**Comment by bjorn3** [comment]
I think just the optimizations it enables are enough to improve binary size. By the way I sent you a PM on zulip in case you missed it.

**Comment by bjorn3** [comment]
Yeah, readonly marking is now correctly done. As for visibility, there doesn't seem to be anything to mark statics as non-exported from the object file. Hidden visibility only prevents it from being exported from dylibs, which rustc handles anyway through version scripts at link time.
